src/http_tcp_server.cpp

```diff
@@ -94,19 +109,57 @@ namespace http
             {
                 exitWithError("Failed to read bytes from client socket connection");
             }
+        
 
             std::ostringstream ss;
             ss << "------ Received Request from client ------\n\n";
             log(ss.str());
 
-            sendResponse();
+            std::string request_string(buffer); //construct string from null-terminated C string.
+            handleRequest(request_string);
 
             close(m_new_socket);
         }
     }
+    void TcpServer::handleRequest(std::string request) {
+        int method_index = request.find(" ");
+        std::string method = request.substr(0, method_index);
+        std::ostringstream ss;
+        ss << "parsed method: " << method << std::endl;
+        log(ss.str());
+        int uri_index = request.find(" ", method_index+1); // grab the string after method up till the next space.
+        std::string uri = request.substr(method_index+1, uri_index-(method_index+1));
+        ss.str("");
+        ss << "parsed URI: '" << uri <<"'"<< std::endl;
+        log(ss.str());        
+        if (method == "GET") { //NOTE: C++ strings implement == operator, can't do in C!
+            handleGetMethod(uri);
+        }
+        else if (method == "HEAD") { //NOTE: C++ strings implement == operator, can't do in C!
+            handleHeadMethod(uri);
+        }
+        else { //501 may be more appropriate here, but this is temporary code.
+            sendResponse(build405Response()); //Method Not Allowed.
+        }
+    }
+    void TcpServer::handleGetMethod(std::string uri) {
+            if (uri == "/index.html" || uri == "/") {
+                sendResponse(buildHomepageResponse(true));
+            }
+            else {
+                sendResponse(build404Response());
+            }
+    }
+    void TcpServer::handleHeadMethod(std::string uri) {
+            if (uri == "/index.html" || uri == "/") {
+                sendResponse(buildHomepageResponse(false));
+            }
+            else {
+                sendResponse(build404Response());
+            }
```

Comment:
Like I mentioned above, we really need one response handler.  I only did individual response handler functions because of how the codebase was originally structured (very strangely), and I wasn't able to change it all overnight.